### PR TITLE
feature: local bootstrap module

### PR DESCRIPTION
**Pull request by jaredfholgate**
## Overview/Summary

This PR adds an option that will just output the files to a directory rather than creating source control. It also optionally deploys the seed resources into Azure.

## This PR fixes/adds/changes/removes

1.  #69 

### Breaking Changes

None, just new functionality

## Testing Evidence

e2e tests updated and run

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/alz-terraform-accelerator/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/alz-terraform-accelerator/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/alz-terraform-accelerator/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by jaredfholgate** [comment]
@luke-taylor This PR has a dependency on https://github.com/Azure/ALZ-PowerShell-Module/pull/96. Once that is released I will re-run the tests.

File: bootstrap/local/variables.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  default     = "basic"
+}
+
+variable "target_directory" {
+  description = "The target directory to create the landing zone files in. (e.g. 'c:\\landingzones\\my_landing_zone')|2"
+  type        = string
+  default     = ""
+}
+
+variable "create_seed_resources_in_azure" {
```
Comment (luke-taylor) [inline bootstrap/local/variables.tf:13]:
```suggestion
variable "create_bootstrap_resources_in_azure" {
```
Just trying to be consistent, we seem to refer to these resources as accelerator resources or bootstrap resources

File: bootstrap/local/main.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+module "files" {
+  source                     = "./../modules/files"
+  starter_module_folder_path = local.starter_module_folder_path
+  additional_files           = var.additional_files
+}
+
+module "azure" {
+  source                             = "./../modules/azure"
+  count                              = var.create_seed_resources_in_azure ? 1 : 0
```
Comment (luke-taylor) [inline bootstrap/local/main.tf:18]:
```suggestion
  count                              = var.create_bootstrap_resources_in_azure ? 1 : 0
```

**Comment by luke-taylor** [comment]
Few semantics, but other than that LGTM

**Comment by jaredfholgate** [comment]
@luke-taylor Tests are passing now. Good to review again.

**Review by luke-taylor — APPROVED** [review]

**Pull request merged**